utils/unittest/logging.go

```diff
@@ -59,16 +60,22 @@ func HookedLogger() (zerolog.Logger, LoggerHook) {
 // logs for testing purposes.
 type LoggerHook struct {
 	logs *strings.Builder
+	mu   *sync.Mutex
 }
 
 // Logs returns the logs as a string
 func (hook LoggerHook) Logs() string {
+	hook.mu.Lock()
+	defer hook.mu.Unlock()
+
 	return hook.logs.String()
 }
 
 // Run implements zerolog.Hook and appends the log message to the log.
 func (hook LoggerHook) Run(_ *zerolog.Event, level zerolog.Level, msg string) {
+	hook.mu.Lock()
 	hook.logs.WriteString(msg)
+	hook.mu.Unlock()
```

Comment:
use defer unlock pattern plz

Reply:
I did not want to hold the lock during `Println`.... let's see, maybe I'll just factor it out into a separate function (it'll be inlined anyway)

Reply:
maybe just print the line before writing to logs?